### PR TITLE
Update system tests to SDK v1.4.0

### PR DESCRIPTION
**Pull request by tomaszslabon**
#Refs: https://github.com/keep-network/tbtc-v2/issues/712.
This PR updates the typescript library to version `1.4.0`.

### CODE REVIEW

File: .github/workflows/system-tests.yml [hunk 1]
```diff
@@ -28,8 +28,7 @@ jobs:
         working-directory: ./solidity
         run: |
           npx hardhat node --no-deploy &
-          HH_NODE_PID=$!
-          while ! lsof -n -Fn -p $HH_NODE_PID | grep -q '^n.*:8545$'; do sleep 10; done
+          while [[ -z $(lsof -i :8545 -t) ]]; do echo "Waiting for port 8545 to be open..."; sleep 10; done
```
Comment (tomaszslabon) [inline .github/workflows/system-tests.yml:31]: When running `npx hardhat node --no-deploy &` on node v18, two processes are spawned (a parent and a child). As a result, the previous solution for waiting for the Hardhat node to start is no longer effective. I've updated the approach: now we simply wait for port `8545` to become available.

File: .github/workflows/system-tests.yml [hunk 2]
```diff
@@ -84,8 +83,7 @@ jobs:
...
```
Comment (tomaszslabon) [inline .github/workflows/system-tests.yml:86]: The same comment as [here](https://github.com/keep-network/tbtc-v2/pull/719/files#r1374357224).

File: .github/workflows/system-tests.yml [hunk 3]
```diff
@@ -116,4 +114,10 @@ jobs:
           WALLET_BITCOIN_WIF: ${{ secrets.SYSTEM_TESTS_WALLET_PRIVATE_KEY_WIF }}
           DEPOSITOR_BITCOIN_WIF: ${{ secrets.SYSTEM_TESTS_DEPOSITOR_PRIVATE_KEY_WIF }}
         run: |
+          # As both tests use the same Bitcoin key and perform the same initial
+          # actions, they often retrieve UTXOs simultaneously. This can lead to
+          # issues where both tests try to spend the same UTXOs, causing
+          # conflicts. Adding a delay helps to mitigate this problem by allowing
+          # time for the UTXO list to update.
+          sleep 240
```
Comment (tomaszslabon) [inline .github/workflows/system-tests.yml:122]: This problem is not strictly related to this PR. It occasionally happened before, for example [in this run](https://github.com/keep-network/tbtc-v2/actions/runs/5773501609/job/15649508601).

File: solidity/hardhat.config.ts [hunk 4]
```diff
@@ -101,7 +101,7 @@ const config: HardhatUserConfig = {
       allowUnlimitedContractSize: process.env.TEST_USE_STUBS_TBTC === "true",
     },
     system_tests: {
-      url: "http://localhost:8545",
+      url: "http://127.0.0.1:8545",
```
Comment (tomaszslabon) [inline solidity/hardhat.config.ts:104]: For some reason `http://localhost:8545` stopped working.
Possibly related to [this](https://stackoverflow.com/questions/68513645/error-connecting-to-localhost-after-npm-hardhat-run).

File: system-tests/package.json [hunk 5]
```diff
@@ -12,14 +12,13 @@
...
-    "test": "hardhat test",
-    "postinstall": "npm rebuild"
+    "test": "hardhat test"
   },
   "files": [
     "test/**/*"
   ],
   "dependencies": {
-    "@keep-network/tbtc-v2.ts": "development",
+    "@keep-network/tbtc-v2.ts": "2.0.0-dev.0",
```
Comment (tomaszslabon) [inline system-tests/package.json:21]: @lukasz-zimnoch 
Should we bring back the `development` tag and simply keep the changes in `yarn.lock`?

Comment (lukasz-zimnoch) [reply system-tests/package.json:21]: Good question! I expect we'll stick to `^2.0.0` while working on https://github.com/keep-network/tbtc-v2/issues/716 so system tests can use all compatible `2.x.x` versions. For now, I'd stick to `2.0.0-dev.0` to explicitly mark the used version.

File: .github/workflows/system-tests.yml (hunk 1 above)
Comment (lukasz-zimnoch) [reply .github/workflows/system-tests.yml:31]: Makes sense!

File: .github/workflows/system-tests.yml [hunk 6]
```diff
@@ -116,4 +114,10 @@ jobs:
           WALLET_BITCOIN_WIF: ${{ secrets.SYSTEM_TESTS_WALLET_PRIVATE_KEY_WIF }}
           DEPOSITOR_BITCOIN_WIF: ${{ secrets.SYSTEM_TESTS_DEPOSITOR_PRIVATE_KEY_WIF }}
         run: |
+          # As system tests use the same Bitcoin key and perform the same initial
+          # actions, they often retrieve UTXOs simultaneously. This can lead to
+          # issues where both tests try to spend the same UTXOs, causing
+          # conflicts. Adding a delay helps to mitigate this problem by allowing
+          # time for the UTXO list to update.
+          sleep 240
```
Comment (lukasz-zimnoch) [inline .github/workflows/system-tests.yml:122]: Maybe an idea is making those tests sequential? We can do it by adding `needs: system-tests-deposit-redemption` on top of the `system-tests-minting-unminting` job definition. I know the execution time will be affected but, if those tests affect each other, this may be a good move.

File: system-tests/package.json [hunk 7]
```diff
@@ -42,14 +41,14 @@
...
+    "hardhat": "^2.18.2",
     "hardhat-deploy": "^0.11.10",
     "mocha": "^9.0.2",
     "prettier": "^2.3.2",
     "ts-node": "^10.1.0",
     "typescript": "^4.3.5"
   },
   "engines": {
-    "node": ">= 14.0.0"
+    "node": ">= 16.0.0"
```
Comment (lukasz-zimnoch) [inline system-tests/package.json:52]: Let's make it simple:
```suggestion
    "node": ">= 16"
```

Comment (tomaszslabon) [reply system-tests/package.json]: Done in 6d6d5b49f979771c0a85d4a7134b10bee058538a.

File: .github/workflows/system-tests.yml (hunk 6 above)
Comment (tomaszslabon) [reply .github/workflows/system-tests.yml]: Done in 0c183c3c580fd584945d79c258dd74a610467ec3.

**Review by lukasz-zimnoch — APPROVED** [review]
